src/Deserialization/BigNum.purs

```diff
@@ -1,14 +1,34 @@
 module Deserialization.BigNum
   ( bigNumToBigInt
+  , bigNumToBigInt'
+  , bigNumToInt
+  , bigNumToInt'
   ) where
 
 import Prelude
 
 import Data.BigInt (BigInt, fromString)
+import Data.Int as Int
 import Data.Maybe (Maybe)
+import Deserialization.Error (FromCslRepError, fromCslRepError)
+import Error (E, noteE)
 import Serialization.Types (BigNum)
+import Type.Row (type (+))
 
 foreign import bigNumToString :: BigNum -> String
 
 bigNumToBigInt :: BigNum -> Maybe BigInt
 bigNumToBigInt = bigNumToString >>> fromString
+
+bigNumToBigInt' :: forall r. String -> BigNum -> E (FromCslRepError + r) BigInt
```

Comment:
I see why it's convenient to use functions like this one, but can we perhaps make a more generic function that prepends a given `String` to error message inside `E (something + r) a`?

Pros: separation of concerns, no need to duplicate each function.

Reply:
I agree that this is a little bit ugly. 
The upside is that it is more convenient to just put any meaningful string instead of "", than to wrap it in `addErrTrace`-like functions.
I'd change it but I don't have time to do it currently. Feel free to revise these, and take a look at `addErrTrace`.